ethsigner/commandline/src/test/java/tech/pegasys/ethsigner/CommandlineParserDownstreamTlsOptionsTest.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.ethsigner.CmdlineHelpers.removeFieldFrom;
+import static tech.pegasys.ethsigner.CmdlineHelpers.validBaseCommandOptions;
+
+import tech.pegasys.ethsigner.core.config.DownstreamTlsOptions;
+import tech.pegasys.ethsigner.core.config.DownstreamTrustOptions;
+import tech.pegasys.ethsigner.core.config.PkcsStoreConfig;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import picocli.CommandLine;
+
+class CommandlineParserDownstreamTlsOptionsTest {
+
+  private final StringWriter commandStdOutput = new StringWriter();
+  private final StringWriter commandErrOutput = new StringWriter();
+  private final PrintWriter stdOut = new PrintWriter(commandStdOutput, true);
+  private final PrintWriter stdErr = new PrintWriter(commandErrOutput, true);
+
+  private EthSignerBaseCommand config;
+  private CommandlineParser parser;
+  private NullSignerSubCommand subCommand;
+  private String defaultUsageText;
+
+  @BeforeEach
+  void setup() {
+    subCommand = new NullSignerSubCommand();
+    config = new EthSignerBaseCommand();
+    parser = new CommandlineParser(config, stdOut, stdErr);
+    parser.registerSigner(subCommand);
+
+    final CommandLine commandLine = new CommandLine(new EthSignerBaseCommand());
+    commandLine.addSubcommand(subCommand.getCommandName(), subCommand);
+    defaultUsageText = commandLine.getUsageMessage();
+  }
+
+  private void missingParameterShowsError(final String input, final String... paramsToRemove) {
```

Comment:
nit: this feels familiar, is it somewhere else we can reuse?

Reply:
yes, as this class was copied from `CommandlineParserTest`, the method gets carried over as well. It should be moved to a generic utility class.

Reply:
extracted out to a utility class,.